File: pawrank/tests/run_all.py
import subprocess, sys, os

tests = ["test_dogs.py"]
all_passed = True
results = []

print("PawRank Test Suite\n")

for test in tests:
    filepath = os.path.join(os.path.dirname(os.path.abspath(__file__)), test)
    try:
        r = subprocess.run([sys.executable, filepath], capture_output=True, text=True, timeout=30)
        print(r.stdout, end="")
        if r.stderr: print(r.stderr, end="")
        results.append((test, "PASS" if r.returncode == 0 else "FAIL"))
        if r.returncode != 0: all_passed = False
    except Exception as e:
        print(f"  ERROR: {test}: {e}")
        results.append((test, "ERROR"))
        all_passed = False

print("\n=== Summary ===")
for name, status in results:
    icon = "\u2705" if status == "PASS" else "\u274c"
    print(f"  {icon} {name}")

p = sum(1 for _, s in results if s == "PASS")
print(f"\n{p}/{len(results)} test suites passed")
sys.exit(0 if all_passed else 1)

File: fitcoin/tests/run-all.js
const { execSync } = require('child_process');
const path = require('path');
const tests = ['portfolio.test.js', 'chart.test.js'];
let allPassed = true;
const results = [];

console.log('FitCoin Test Suite\n');

tests.forEach(test => {
  try {
    const output = execSync(`node "${path.join(__dirname, test)}"`, { encoding: 'utf8', stdio: 'pipe' });
    console.log(output);
    results.push({ test, status: 'PASS' });
  } catch (err) {
    console.log(err.stdout || '');
    console.error(err.stderr || '');
    results.push({ test, status: 'FAIL' });
    allPassed = false;
  }
});

console.log('\n=== Summary ===');
results.forEach(r => {
  const icon = r.status === 'PASS' ? '\u2705' : '\u274c';
  console.log(`  ${icon} ${r.test}`);
});
const p = results.filter(r => r.status === 'PASS').length;
console.log(`\n${p}/${results.length} test suites passed`);
process.exit(allPassed ? 0 : 1);
